src/main/java/de/tum/in/www1/artemis/service/StatisticsService.java

```diff
@@ -0,0 +1,140 @@
+package de.tum.in.www1.artemis.service;
+
+import java.time.YearMonth;
+import java.time.ZonedDateTime;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.springframework.stereotype.Service;
+
+import de.tum.in.www1.artemis.domain.enumeration.SpanType;
+import de.tum.in.www1.artemis.repository.StatisticsRepository;
+
+@Service
+public class StatisticsService {
+
+    private final StatisticsRepository statisticsRepository;
+
+    public StatisticsService(StatisticsRepository statisticsRepository) {
+        this.statisticsRepository = statisticsRepository;
+    }
+
+    public Integer getLoggedInUsers(Long span) {
+        return this.statisticsRepository.getLoggedInUsers(ZonedDateTime.now().minusDays(span).toInstant());
+    }
+
+    public Integer getActiveUsers(Long span) {
+        return this.statisticsRepository.getActiveUsers(ZonedDateTime.now().minusDays(span));
+    }
+
+    /**
+     * Forwards the request to the repository, which returns a List<Map<String, Object>>, with String being the column name, "day" and "amount" and Object being the value,
+     * either the date or the amount of submissions. It then collects the amounts in an array, depending on the span value, and returns it
+     *
+     * @param span DAY,WEEK,MONTH or YEAR depending on the active tab in the view
+     * @return a array, containing the values for each bar in the graph
+     */
+    public Integer[] getTotalSubmissions(SpanType span) {
```

Comment:
Holy that's a really big and complex method. Is there a way to break this down into smaller functions and/or define queries that will do the heavy work for you?

Reply:
It's really hard to understand what's happening here. If it's impossible to break this down then I'd suggest writing comments so that developers can understand this quicker.

Reply:
Yes you are right. It is a pretty complex method due to the fact that every SpanType needs to get handled seperately. I tried to split it up in smaller methods now and commented it in a pretty detailed way. Hope this is better. Done in [this](https://github.com/ls1intum/Artemis/pull/2471/commits/5905879e8731f464bb3777cae0b6e686a48611d8) commit